### DISCUSSION TITLE
M1 Mac: zsh segmentation fault  pyarmor reg

### DISCUSSION THREAD

**Discussion by Kaiyangshi-Ito** [dashingsoft/pyarmor#1155] [Q&A] [answered]
If I type `pyarmor reg pyarmor-regfile-******.zip`, I got:   
```
INFO     Python 3.9.13
INFO     Pyarmor 8.1.1 (trial), 000000, non-profits
INFO     Platform darwin.aarch64
INFO     register "pyarmor-regfile-******.zip"
INFO     update license token
zsh: segmentation fault  pyarmor reg
```
I think it might be an issue with Mac OS using zsh instead of bash?

**Comment by jondy** [comment]
I can't reproduce it in My Apple M1.

It more like code sign issue. Please check it `codesign -v /path/to/pytransform3.so`, if it's not valid, force it with ad-hoc code sign.

**Comment by Kaiyangshi-Ito** [comment]
> I can't reproduce it in My Apple M1.
> 
> It more like code sign issue. Please check it `codesign -v /path/to/pytransform3.so`, if it's not valid, force it with ad-hoc code sign.

Look valid to me, I typed `codesign -v -v /path/to/pytransform3.so`, got:
```
/path/to/pytransform3.so: valid on disk
/path/to/pytransform3.so: satisfies its Designated Requirement
```

**Comment by jondy** [comment, marked as answer]
If code sign is OK, then it's security configuration issue, please console Apple documentation to find solution, it's out of pyarmor.
